src/AbstractZonotope.jl

```diff
@@ -136,3 +136,55 @@ An integer representing the number of generators.
 function ngens(Z::AbstractZonotope)::Int
     return length(generators(Z))
 end
+
+"""
+    minkowski_sum(Z1::AbstractZonotope{N}, Z2::AbstractZonotope{N})
+        where {N<:Real}
+
+Concrete Minkowski sum of a pair of zonotopic sets.
+
+### Input
+
+- `Z1` -- zonotopic set
+- `Z2` -- zonotopic set
+
+### Output
+
+A `Zonotope` corresponding to the concrete Minkowski sum of `Z1` and `Z2`.
+
+### Algorithm
+
+The resulting zonotope is obtained by summing up the centers and concatenating
+the generators of `Z1` and `Z2`.
+"""
+function minkowski_sum(Z1::AbstractZonotope{N},
+                       Z2::AbstractZonotope{N}) where {N<:Real}
+    return Zonotope(center(Z1) + center(Z2), [genmat(Z1) genmat(Z2)])
+end
+
+"""
+    linear_map(M::AbstractMatrix{N}, Z::AbstractZonotope{N}) where {N<:Real}
+
+Concrete linear map of a zonotopic set.
+
+### Input
+
+- `M` -- matrix
+- `Z` -- zonotopic set
+
+### Output
+
+A zonotope corresponding to the concrete linear map `M` applied to `Z`.
+
+### Algorithm
+
+The resulting zonotope is obtained by applying the linear map `M` to the center
+and the generators of `Z`.
+"""
+function linear_map(M::AbstractMatrix{N},
```

Comment:
Dispatching `linear_map` on the `AbstractZonotope` interface has effects on other other sets and we should consider it before making this change (given that `AbstractZonotope <: AbstractPolyhedron`, cf. [this function](https://github.com/JuliaReach/LazySets.jl/blob/master/src/AbstractPolyhedron_functions.jl#L300)).

For instance, in master `linear_map(M, H)` for `H` a hyperrectangle returns a new hyperrectangle, while this branch returns a zonotope. This is not necessarily bad, since linear maps on zonotopes are very fast, maybe even faster than the current implementation ( ;) ), which needs to shift the constraints according to `M`. However, returning an hyperrectangle seems better IMO (independently of the implementation).

Can you dispatch on `Zonotope` here and let the generalization for later?

Reply:
Given that you already did the benchmarks, do you still think we should postpone?

Reply:
The change in `linear_map` was moved to #1461.